permissioned-token/program/src/instruction.rs

```diff
@@ -0,0 +1,215 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use shank::ShankInstruction;
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    system_program, sysvar,
+};
+use spl_associated_token_account::get_associated_token_address;
+
+#[derive(Debug, Clone, ShankInstruction, BorshSerialize, BorshDeserialize)]
+#[rustfmt::skip]
+pub enum PermissionedTokenInstruction {
+
+    #[account(0, writable, signer, name = "mint")]
+    #[account(1, writable, signer, name = "payer")]
+    #[account(2, name = "upstream_authority")]
+    #[account(3, name = "system_program", desc = "System program")]
+    #[account(4, name = "token_program", desc = "Token program")]
+    InitializeMint {
+        decimals: u8,
+    },
+
+    #[account(0, writable, name = "account")]
+    #[account(1, name = "owner")]
+    #[account(2, writable, signer, name = "payer")]
+    #[account(3, signer, name = "upstream_authority")]
+    #[account(4, name = "freeze_authority")]
+    #[account(5, name = "mint")]
+    #[account(6, name = "system_program", desc = "System program")]
+    #[account(7, name = "rent", desc = "Rent sysvar")]
+    #[account(
+        8,
+        name = "associated_token_program",
+        desc = "Associated Token program"
+    )]
+    #[account(9, name = "token_program", desc = "Token program")]
+    InitializeAccount,
+
+    #[account(0, writable, name = "src_account")]
+    #[account(1, writable, name = "dst_account")]
+    #[account(2, name = "mint")]
+    #[account(3, signer, name = "owner")]
+    #[account(4, signer, name = "upstream_authority")]
+    #[account(5, name = "freeze_authority")]
+    #[account(6, name = "token_program", desc = "Token program")]
+    Transfer { amount: u64 },
+
+    #[account(0, writable, name = "mint")]
+    #[account(1, writable, name = "account")]
+    #[account(2, signer, name = "mint_authority")]
+    #[account(3, signer, name = "upstream_authority")]
+    #[account(4, name = "freeze_authority")]
+    #[account(5, name = "token_program", desc = "Token program")]
+    MintTo { amount: u64 },
+
+    #[account(0, writable, name = "mint")]
+    #[account(1, writable, name = "account")]
+    #[account(2, signer, name = "owner")]
+    #[account(3, signer, name = "upstream_authority")]
+    #[account(4, signer, name = "freeze_authority")]
```

Comment:
looks like this one got missed:

```suggestion
    #[account(4, name = "freeze_authority")]
```